src/Dijkstra/AST/Syntax.agda

```diff
@@ -6,15 +6,39 @@
 
 module Dijkstra.AST.Syntax where
 
-open import Data.Empty
-open import Data.Product using (_×_ ; _,_ ; proj₁ ; proj₂)
-open import Data.Unit
-open import Function
-open import Haskell.Prelude
+open import Level renaming (_⊔_ to _ℓ⊔_ ; suc to ℓ+1)
+
+record Functor  {ℓ₁ ℓ₂ : Level} (F : Set ℓ₁ → Set ℓ₂) : Set (ℓ₂ ℓ⊔ ℓ+1 ℓ₁) where
```

Comment:
Functor, Applicative and Monad should go into a separate file (like we did in under `Haskell`.